osu.Game.Rulesets.Mania/Difficulty/ManiaPerformanceCalculator.cs

```diff
@@ -34,82 +32,43 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
 
             double difficultyValue = computeDifficultyValue(maniaAttributes);
-            double accValue = computeAccuracyValue(difficultyValue, maniaAttributes);
-            double totalValue =
-                Math.Pow(
-                    Math.Pow(difficultyValue, 1.1) +
-                    Math.Pow(accValue, 1.1), 1.0 / 1.1
-                ) * multiplier;
+            double totalValue = difficultyValue * multiplier;
 
             return new ManiaPerformanceAttributes
             {
                 Difficulty = difficultyValue,
-                Accuracy = accValue,
-                ScaledScore = scaledScore,
                 Total = totalValue
             };
         }
 
         private double computeDifficultyValue(ManiaDifficultyAttributes attributes)
         {
-            double difficultyValue = Math.Pow(5 * Math.Max(1, attributes.StarRating / 0.2) - 4.0, 2.2) / 135.0;
-
-            difficultyValue *= 1.0 + 0.1 * Math.Min(1.0, totalHits / 1500.0);
-
-            if (scaledScore <= 500000)
-                difficultyValue = 0;
-            else if (scaledScore <= 600000)
-                difficultyValue *= (scaledScore - 500000) / 100000 * 0.3;
-            else if (scaledScore <= 700000)
-                difficultyValue *= 0.3 + (scaledScore - 600000) / 100000 * 0.25;
-            else if (scaledScore <= 800000)
-                difficultyValue *= 0.55 + (scaledScore - 700000) / 100000 * 0.20;
-            else if (scaledScore <= 900000)
-                difficultyValue *= 0.75 + (scaledScore - 800000) / 100000 * 0.15;
-            else
-                difficultyValue *= 0.90 + (scaledScore - 900000) / 100000 * 0.1;
+            double difficultyValue = Math.Pow(Math.Max(attributes.StarRating - 0.15, 0.05), 2.2) // Star rating to pp curve
+                                     * Math.Max(0, 5 * scoreAccuracy - 4) // Accuracy curve
+                                     * (1 + 0.1 * Math.Min(1, totalHits / 1500)); // Length bonus
 
             return difficultyValue;
         }
 
-        private double computeAccuracyValue(double difficultyValue, ManiaDifficultyAttributes attributes)
-        {
-            if (attributes.GreatHitWindow <= 0)
-                return 0;
-
-            // Lots of arbitrary values from testing.
-            // Considering to use derivation from perfect accuracy in a probabilistic manner - assume normal distribution
-            double accuracyValue = Math.Max(0.0, 0.2 - (attributes.GreatHitWindow - 34) * 0.006667)
-                                   * difficultyValue
-                                   * Math.Pow(Math.Max(0.0, scaledScore - 960000) / 40000, 1.1);
-
-            return accuracyValue;
-        }
-
         private double totalHits => countPerfect + countOk + countGreat + countGood + countMeh + countMiss;
+        private double customAccuracy => (countPerfect * 320 + countGreat * 300 + countGood * 200 + countOk * 100 + countMeh * 50) / (totalHits * 320);
```

Comment:
Is this intentionally meant to differ from the score's accuracy (`score.Accuracy`)? As far as I'm seeing here, this is pretty much the same except for perfect judgements being computed as awarding 320 instead of [what they actually award (315)](https://github.com/ppy/osu/blob/b29172ea45fea63d8e48b85aee346d9d607f0dfa/osu.Game/Rulesets/Judgements/Judgement.cs#L162-L163).

Reply:
Yes. The purpose of this being its own thing is so judgements can be weighted independent of accuracy. After we see the pp changes on smoogi's sheet, this will probably be adjusted further, possibly by further increasing the weight of Perfect or weighting Miss as a negative value. 320 was chosen as that was my initial value for balancing.